pipeline/backend/mock/mock.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2024 Woodpecker Authors
```

Comment:
Why not using mockery to generate a mockable backend?

Reply:
because that can only be used in unit tests ... if you want integration tests it has to be standalone

so you can not generate and mock it within the go code you test against.

If we need a mocked backend for unit test ... we of course can use mockery ....

:thinking: we might rename it to `dummy`